src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -188,7 +178,28 @@ class OrderedRVD private(
       return this
     if (shuffle) {
       val shuffled = rdd.coalesce(maxPartitions, shuffle = true)
-      val ranges = OrderedRVD.calculateKeyRanges(typ, OrderedRVD.getPartitionKeyInfo(typ, shuffled), shuffled.getNumPartitions)
+
+      val localKType = typ.kType
+      val localKIndex = typ.kRowFieldIdx
+      val localRowType = typ.rowType
+      val shuffledKeyRDD = shuffled.mapPartitions { it =>
+        val rv2 = RegionValue()
+        val rvb = new RegionValueBuilder()
+        it.map { rv =>
+          rvb.set(rv.region)
+          rvb.start(localKType)
+          rvb.startStruct()
+          var i = 0
+          while (i < localKType.size) {
```

Comment:
This can be RVB.addFields, right?

Reply:
I don't think so. The struct on the inside is localRowType, but I only want to add the key fields